### PR TITLE
[JIT] Bind AliasInfo to decrease differences in interfaces across languages

### PR DESCRIPTION
**Pull request by ttanpcs**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79814
* __->__ #79661
* #79474

- Changes python bindings to derive from AliasInfo class instead of Argument Class so that the C++ and python interfaces are more uniform

### CODE REVIEW

File: test/test_public_bindings.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def test_no_new_bindings(self):
         torch_C_allowlist_superset = {
             "AggregationType",
             "AliasDb",
+            "AliasInfo"
```
Comment (albanD) [inline test/test_public_bindings.py:36]: Do we really want `torch.AliasInfo` to be a thing? Is this new public API properly documented?

File: torch/csrc/jit/python/init.cpp [hunk 2]
```diff
@@ -1490,42 +1490,32 @@ void initJITBindings(PyObject* module) {
...
             return self.default_value().has_value();
           })
+      .def_property_readonly(
+          "alias_info", [](Argument& self) { return self.alias_info(); })
       .def_property_readonly(
           "is_out", [](Argument& self) { return self.is_out(); })
+      .def_property_readonly("kwarg_only", [](Argument& self) -> bool {
+        return self.kwarg_only();
+      });
+  py::class_<AliasInfo>(m, "AliasInfo")
```
Comment (davidberard98) [inline torch/csrc/jit/python/init.cpp:1500]: You can probably make this "_AliasInfo" and then you won't need to add it into the torch_C_allowlist_superset

File: test/test_public_bindings.py (hunk 1 above)
Comment (davidberard98) [reply test/test_public_bindings.py]: changed it to torch._C._AliasInfo so this won't be an issue

**Review by davidberard98 — APPROVED** [review]
LGTM!

**Pull request closed without merging**